Classes/Extension/ExtensionSetupRenderer.php

```diff
@@ -0,0 +1,122 @@
+<?php
+namespace Helhum\Typo3Console\Extension;
+
+/*
+ * This file is part of the TYPO3 Console project.
+ *
+ * It is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU General Public License, either version 2
+ * of the License, or any later version.
+ *
+ * For the full copyright and license information, please read
+ * LICENSE file that was distributed with this source code.
+ *
+ */
+
+use Helhum\Typo3Console\Database\Schema\SchemaUpdateResult;
+use Helhum\Typo3Console\Database\Schema\SchemaUpdateType;
+use TYPO3\CMS\Core\Utility\GeneralUtility;
+use TYPO3\CMS\Extbase\SignalSlot\Dispatcher;
+use TYPO3\CMS\Extensionmanager\Utility\InstallUtility;
+
+/**
+ * Class ExtensionSetup
+ */
+class ExtensionSetupRenderer
+{
+    /**
+     * @var \Helhum\Typo3Console\Mvc\Cli\ConsoleOutput
+     * @inject
+     */
+    protected $output;
+
+    /**
+     * @var \Helhum\Typo3Console\Database\Schema\SchemaUpdateResultRenderer
+     * @inject
+     */
+    protected $schemaUpdateResultRenderer;
```

Comment:
please use constructor injection and at least do not make output optional. The currently in use ConsoleOutput object should be passed instead of creating a new one.
better pass both deps on object creation

ExtensionSetupRenderer should be created/ used in the CommandController

add signal slot dispatcher as dep as well in the constructor